### PR TITLE
ProductList and State Management

### PR DESCRIPTION
**Pull request by Jose-Hash-Dev**
ProductList Update and State Management implementation with ReactJs createContext

### CODE REVIEW

**Comment by Jose-Hash-Dev** [comment]
Capture

**Comment by Jose-Hash-Dev** [comment]
- Used createContext and useReducer from React instead of Redux
- Changed ProductList UI and added additional functionalities
- User can add product from ProductList
- User can see if product is already in the cart or not
- Data is stored in MongoDB using Mongoose ODM

File: pages/api/products/index.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import Book from "../../../models/Book";
+import dbConnect from "../../../lib/dbConnect";
+dbConnect().then((result) => {
+  console.log("Connection in index.tsx");
```
Comment (Jalal999) [inline pages/api/products/index.ts:5]: do you need that or it is just for checking purposes?

File: pages/api/products/index.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+export default async function getAllProducts(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === "GET") {
+    try {
+      const books = await Book.find();
+      res.status(200).json(books);
+    } catch (err) {
+      res.status(500).json(err);
```
Comment (Jalal999) [inline pages/api/products/index.ts:17]: what if you get different error type from response? will this status code of 500 work for all or not?

File: pages/api/products/index.ts (hunk 1 above)
Comment (Jose-Hash-Dev) [reply pages/api/products/index.ts:5]: It was for testing purposes, will delete for next PR

File: pages/api/products/index.ts (hunk 2 above)
Comment (Jose-Hash-Dev) [reply pages/api/products/index.ts:17]: `status(500)` is internal server error I used for all errors during `get request`. Different codes can be used but usually the error can be checked from terminal. Generally error is related to connection issues to MongoDB. For example fetching data from MongoDB before connection is established will throw this error.

**Pull request merged**